apps/prairielearn/src/migrations/20250707220720_enrollments__meta_columns__add.sql

```diff
@@ -0,0 +1,39 @@
+-- Add new enum type for enrollment status if not exists
+DO $$
+BEGIN
+    IF NOT EXISTS (SELECT 1 FROM pg_type WHERE typname = 'enum_enrollment_status') THEN
+        CREATE TYPE enum_enrollment_status AS ENUM ('invited', 'joined', 'removed', 'blocked');
```

Comment:
In PT, we have a `rejected` status for someone who rejected an invite. Might we want the same here? This is potentially relevant if we want more constraints; we'd want a non-invited state where `pending_uid` is nullable.

Did we ever talk about if invites would work the same way as they did in PT? That is, where students have to actually accept invites? I think we did, and I think we do want that behavior, but I'm not sure if our overarching plan is written down anywhere so I wasn't sure.

Reply:
Does it make sense to instead have a `invited` status with a deleted_at field?